### PR TITLE
[PORT] The Hugbot 

### PR DESCRIPTION
**Pull request by Bobbanz1**
## About The Pull Request
Completes one of the suggestions made on the forum: 
https://forums-nsv.beestation13.com/t/new-bots-title-needs-to-be-15-characters-soooooooo/183

This PR accomplishes that by porting the Hugbot from Skyrat13
In order to build a Hugbot you need 1 borg arm, attach it to a box of hugs(found in a hacked medical vending machine) and then add a proximity sensor to it. 

The Hugbot, as its name implies, goes about hugging crew members. It will however not hug crew members which have already got the hugged moodlet. 

The Hugbot has adjustable operating modes, allowing it to hug, pat and boop the snoot of the crew! ~~**HEADPATS FOR EVERYONE!!**~~

The Hugbot will provide helpful and friendly moral support to the crew by giving them friendly reminders that they are an important member of the crew, and that they are always loved.

If emagged, the Hugbot will hug a bit too hard, knock the crew down and replace it's friendly vocabulary with hostile statements.
## Ports From: Skyrat13
* https://github.com/Skyrat-SS13/Skyrat13/pull/3572

## Why It's Good For The Game
~~Furthers the Organic Replacement with Machines by replacing the Clown's job with a mindless Robot that hugs people~~
Adds a new Bot to the Game, one that can give you a peptalk and help keep the crews spirit up in the face of an all out space war

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://user-images.githubusercontent.com/59128051/224109895-c14524fa-c052-47f2-8dfc-3e1692fa9676.mp4

</details>

## Changelog
:cl:jbox144, Bobbanz1
add: Adds the hugbot
imageadd: Adds icons for the hugbot
imageadd: Adds the NSV13 file aibots.dmi
imagedel: Deleted the catmedbot.dmi file
code: Changed the icon path on the MediKitty to the newly created dmi file
/:cl:

### CODE REVIEW

File: code/__DEFINES/robots.dm [hunk 1]
```diff
@@ -36,6 +36,7 @@
 #define MED_BOT				(1<<4)	//!  Medibots
 #define HONK_BOT			(1<<5)	//!  Honkbots & ED-Honks
 #define FIRE_BOT			(1<<6)  //!  Firebots
+#define HUG_BOT 			(1<<7)  //!  Hugbots //NSV13
```
Comment (Kmc2000) [inline code/__DEFINES/robots.dm:39]: I would suggest this goes in a modular file, if this is not possible I'm OK with taking this as is.

File: nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+		path = list()
+		mode = BOT_IDLE
+		last_found = world.time
+
+	else if(stationary_mode && patient) //Since we cannot move in this mode, ignore the patient and wait for another.
+		soft_reset()
+		return
+
+	if(patient && path.len == 0 && (get_dist(src,patient) > 1))
+		path = get_path_to(src, get_turf(patient), /turf/proc/Distance_cardinal, 0, 30,id=access_card)
```
Comment (covertcorvid) [inline nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm:310]:
```suggestion
		path = get_path_to(src, patient, 30,id=access_card)
```
Our get_path_to doesn't need a proc definition and will do the get_turf itself

File: nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+		mode = BOT_MOVING
+		if(!path.len) //try to get closer if you can't reach the patient directly
+			path = get_path_to(src, get_turf(patient), /turf/proc/Distance_cardinal, 0, 30,1,id=access_card)
```
Comment (covertcorvid) [inline nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm:313]:
```suggestion
			path = get_path_to(src, patient, 30,1,id=access_card)
```

File: nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+		oldpatient = patient
+		soft_reset()
+		return
+
+	if (get_dist(src, H) > 1)
+		mode = BOT_MOVING
+		update_icon()
+		return
+
+	// Hey, didyah know that you Han't hug people as non-carbon? Well you can't hug people as non-carbon. Hence this mess
```
Comment (covertcorvid) [inline nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm:360]: fear

File: nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm [hunk 5]
```diff
@@ -0,0 +1,515 @@
...
+		SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "headpat", /datum/mood_event/headpat)
+	else if(check_zone(zone_selected) == BODY_ZONE_HEAD)
+		var/datum/species/S
+		if(ishuman(H))
+			S = H.dna.species
+
+			visible_message("<span class='notice'>[src] gives [H] a pat on the head to make [H.p_them()] feel better!</span>", \
+						"<span class='notice'>You give [src] a pat on the head to make [H.p_them()] feel better!</span>")
+			SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "headpat", /datum/mood_event/headpat)
+			if(S?.can_wag_tail(src) && !H.dna.species.is_wagging_tail())
```
Comment (covertcorvid) [inline nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm:375]:
```suggestion
			if(S?.can_wag_tail(src) && !(S?.is_wagging_tail()))
```

File: nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm [hunk 6]
```diff
@@ -0,0 +1,515 @@
...
+				var/static/list/many_tails = list("tail_human", "tail_lizard", "mam_tail")
```
Comment (covertcorvid) [inline nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm:376]:
```suggestion
				var/static/list/many_tails
				if(!many_tails)
					many_tails = list("tail_human", "tail_lizard", "mam_tail")
```

**Review by covertcorvid — CHANGES_REQUESTED** [review]

File: nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm [hunk 7]
```diff
@@ -0,0 +1,515 @@
...
+	model = "Hugbot"
+	bot_core_type = /obj/machinery/bot_core/hugbot
+	window_id = "autohug"
+	window_name = "Automatic Hugging Unit v2.0"
+	path_image_color = "#FFDDDD"
+
+	var/stationary_mode = 0 //If enabled, the Hugbot will not move automatically.
+	var/max_sanity = 70 // Sanity needed to allow hugging
+	var/shut_up = 0 // Self explanitory
+	var/tania_mode = 0 // When enabled, the hug bot will not change targets, and will hug regardless of mood
```
[the remark this answers is not in the document]
Comment (Bobbanz1) [reply nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm:39]: I have no idea really, could try hunting down the original PR creator but I haven't got the faintest idea where I'd even start.

Could probably be a reference to something but a quick google search reveals nothing.

[the remark this answers is not in the document]
Comment (covertcorvid) [reply nsv13/code/modules/mob/living/simple_animal/bot/hugbot.dm:39]: You can leave it, I was just curious. Not curious enough for you to go to the effort of hunting someone down

**Review by covertcorvid — APPROVED** [review]
o7

**Pull request merged**